docs/style.md

```diff
@@ -0,0 +1,107 @@
+# Style guide
+
+Please adhere to the following style guide when writing code for the RISCy Kernel.
+
+## DRY
+
+**Don't Repeat Yourself.** <br>
+**Don't Repeat Yourself.** <br>
+**Don't Repeat Yourself.**
+
+Its easy to copy-paste. It's also easy to duplicate bugs that way. Just define a `static` functions, even if it is five lines long and takes four arguments. 
+The compiler is smart and will inline it appropriately.
+
+## Language
+
+We're using **C 23** and **C++ 20**. Go crazy, you've got all the new features.
+
+For example, annotate your fallthrough switch cases explicitly:
+```c
+switch (x) {
+    case 1:
+        a = 5;
+        [[fallthrough]];
+    case 2:
+        foo();
+        break;
+}
+```
+
+## Formatting
+
+This is handled by `clang-foramt`. Just run `make format` and your code will become pretty.
+
+Some special cases are listed below.
+
+### Single-line `if` statements
+
+Feel free to drop the branches on single-line `if` statements. We avoid bugs such as the infamous
```

Comment:
I hate this

Reply:
I agree with @echavemann

Reply:
Single line "if"'s are awful

Reply:
Linux kernel style guide

![image](https://github.com/riscy-development/riscy-os/assets/76668284/eb67b3f9-2647-43ab-be0c-5f2871430caf)

Reply:
I know and I cringe every time I'm reading linux and see this

Reply:
Single line `if`s are awesome

otherwise you end up with `}` littered around your code

it also helps with single-line if-else statements

```c
if (x)
    foo();
else
    bar();
```

is much easier to read than

```c
if (x) {
    foo();
}
else {
    bar();
}
```

Reply:
Its like ternaries

Reply:
hmmmm, I'm not going to use them in all likelihood ever. (And if you want to modify code that @echavemann or I write to use them do it in a separate commit don't make it part of PR reviews) But as long as you agree to both of those conditions I'm okay if you use them.

Reply:
(If you can just get the formatter to do it automatically (sensibly) then I'm fine with that too)

Reply:
no my reasoning here is

"feel free to use them if you want"
"but if you don't its also ok"